src/hotspot/share/oops/method.cpp

```diff
@@ -2187,7 +2187,7 @@ void Method::ensure_jmethod_ids(ClassLoaderData* loader_data, int capacity) {
     // Have to add jmethod_ids() to class loader data thread-safely.
     // Also have to add the method to the list safely, which the cld lock
     // protects as well.
-    MutexLocker ml(cld->metaspace_lock(),  Mutex::_no_safepoint_check_flag);
+    MutexLocker ml(JmethodIdCreation_lock,  Mutex::_no_safepoint_check_flag);
```

Comment:
The comment above this states:
 // Also have to add the method to the list safely, which the cld lock
 // protects as well.
I don't know what list is being referred to here but either the comment is incorrect or else the change of lock is incorrect!

Reply:
We create the list with:
`cld->set_jmethod_ids(new JNIMethodBlock(capacity));`
And add to the list:
`return (jmethodID)cld->jmethod_ids()->add_method(m);`

I removed 'cld' from comment.